File: library/src/main/java/me/shkschneider/skeleton/ui/Snack.java
package me.shkschneider.skeleton.ui;

import android.support.annotation.NonNull;
import android.support.design.widget.Snackbar;
import android.text.TextUtils;
import android.view.View;

import me.shkschneider.skeleton.helper.LogHelper;

public class Snack {

    public static void bar(@NonNull final View view, @NonNull final String msg) {
        bar(view, msg, null, null);
    }

    public static void bar(@NonNull final View view, @NonNull final String msg, final String action, final View.OnClickListener onClickListener) {
        if (TextUtils.isEmpty(msg)) {
            LogHelper.warning("Message was NULL");
        }
        final Snackbar snackbar = Snackbar.make(view, msg, Snackbar.LENGTH_SHORT);
        if (! TextUtils.isEmpty(action) && onClickListener != null) {
            snackbar.setAction(action, onClickListener);
        }
        snackbar.show();
    }

}

File: library/src/main/java/me/shkschneider/skeleton/ui/Snack.kt
package me.shkschneider.skeleton.ui

import android.view.View
import androidx.annotation.UiThread
import com.google.android.material.snackbar.Snackbar
import me.shkschneider.skeleton.extensions.then

import me.shkschneider.skeleton.helper.Logger

object Snack {

    @UiThread
    fun bar(view: View, msg: String, action: String? = null, onClickListener: View.OnClickListener? = null, duration: Int? = null) {
        if (msg.isBlank()) {
            Logger.warning("Message was NULL")
        }
        val snackBar = Snackbar.make(view, msg, Snackbar.LENGTH_SHORT)
        onClickListener?.let {
            snackBar.setAction(action, it)
        }
        snackBar.duration = duration ?: (action.isNullOrBlank()) then Snackbar.LENGTH_SHORT ?: Snackbar.LENGTH_INDEFINITE
        snackBar.show()
    }

}
